compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUCastTypeToFitMMA.cpp

```diff
@@ -89,17 +89,21 @@ struct LLVMGPUCastTypeToFitMMAPass
         func->getAttrOfType<IREE::GPU::MMAScheduleAttr>(scheduleAttrName);
     if (!scheduleAttr) {
       DictionaryAttr configDict = getTranslationInfo(func).getConfiguration();
-      scheduleAttr = dyn_cast_or_null<IREE::GPU::MMAScheduleAttr>(
-          configDict.get(scheduleAttrName));
+      if (configDict) {
+        scheduleAttr = dyn_cast_or_null<IREE::GPU::MMAScheduleAttr>(
+            configDict.get(scheduleAttrName));
+      }
     }
-    if (!scheduleAttr) {
-      func.emitError() << "missing mma_schedule\n";
-      return signalPassFailure();
+
+    if (scheduleAttr) {
+      func.walk([&](vector::ContractionOp contract) {
+        contract->setAttr("iree.amdgpu.mma", scheduleAttr.getIntrinsic());
```

Comment:
nit: I would check if the contract already has this attribute set and only set it if not present. Also maybe add a comment
```
// Import mma type from dispatch schedule attribute if present.
```

Reply:
That makes sense :)